### ISSUE TITLE
Export pipelines

### ISSUE THREAD

**Issue by emanuelqueiroga** [alteryx/evalml#2081]
I'm new to using the library. I found EvalML to be very interesting and I am trying to learn more about it. At the moment I use TPOT in my projects, it is possible to export a pipeline ready for use and consequent export of the model. I would like to clarify a doubt, is there anything of the kind in EvalML?

**Comment by dsherry** [comment]
@emanuelqueiroga thanks for your interest!

If you're asking about how to pick one of the models from an automl search and export it: yes you can do this. Check out the call to `get_pipeline` at the bottom of [our start guide](https://evalml.alteryx.com/en/stable/start.html). You can fit that pipeline on your entire training data and then use it to generate predictions.

If you're asking if our pipeline datastructure can be easily saved and loaded in another python runtime: yes you can do this. We don't mention this in our user guide yet, but I just filed an issue to fix that ( #2083 ). Here's an example of how you'd do it:
```python
import evalml
X, y = evalml.demos.load_breast_cancer()
class MyBinaryClassificationPipeline(BinaryClassificationPipeline):
    component_graph = ['Imputer', 'Random Forest Classifier']
pipeline = MyBinaryClassificationPipeline({})
pipeline.fit(X, y)
pipeline.save('~/pipeline.cloudpickle')

# then in another python process you can load the pipeline and use it
pipeline = evalml.pipelines.PipelineBase.load('~/pipeline.cloudpickle')
pipeline.predict(X_new)
```

For more info check out [our user guide on our pipeline datastructure](https://evalml.alteryx.com/en/stable/user_guide/pipelines.html).
And here's our API reference on the pipeline [save](https://evalml.alteryx.com/en/stable/generated/methods/evalml.pipelines.PipelineBase.save.html#evalml.pipelines.PipelineBase.save) and [load](https://evalml.alteryx.com/en/stable/generated/methods/evalml.pipelines.PipelineBase.load.html#evalml.pipelines.PipelineBase.load) methods

Does this answer your question?

**Comment by emanuelqueiroga** [comment]
> @emanuelqueiroga thanks for your interest!
> 
> If you're asking about how to pick one of the models from an automl search and export it: yes you can do this. Check out the call to `get_pipeline` at the bottom of [our start guide](https://evalml.alteryx.com/en/stable/start.html). You can fit that pipeline on your entire training data and then use it to generate predictions.
> 
> If you're asking if our pipeline datastructure can be easily saved and loaded in another python runtime: yes you can do this. We don't mention this in our user guide yet, but I just filed an issue to fix that ( #2083 ). Here's an example of how you'd do it:
> 
> ```python
> import evalml
> X, y = evalml.demos.load_breast_cancer()
> class MyBinaryClassificationPipeline(BinaryClassificationPipeline):
>     component_graph = ['Imputer', 'Random Forest Classifier']
> pipeline = MyBinaryClassificationPipeline({})
> pipeline.fit(X, y)
> pipeline.save('~/pipeline.cloudpickle')
> 
> # then in another python process you can load the pipeline and use it
> pipeline = evalml.pipelines.PipelineBase.load('~/pipeline.cloudpickle')
> pipeline.predict(X_new)
> ```
> 
> For more info check out [our user guide on our pipeline datastructure](https://evalml.alteryx.com/en/stable/user_guide/pipelines.html).
> And here's our API reference on the pipeline [save](https://evalml.alteryx.com/en/stable/generated/methods/evalml.pipelines.PipelineBase.save.html#evalml.pipelines.PipelineBase.save) and [load](https://evalml.alteryx.com/en/stable/generated/methods/evalml.pipelines.PipelineBase.load.html#evalml.pipelines.PipelineBase.load) methods
> 
> Does this answer your question?

First of all I would like to thank you for your response.
I believe that I did not give enough details about the situation where I want to implement EvalML.
I currently have an academic application running that predicts a particular situation. This web application runs by get a .pkl file and applying it to the data upload by users. At the moment I cannot change the web application because it was developed by another team, so I can only change the .pkl file. My intention is to use EvalML to generate a pipeline that I can use to manually generate a new .pkl file with the retraining for that application. I thought of something like exporting a TPOT-style pipeline, which is basically a SkLearn pipeline and I adapt it for my data. However, I have not found the solution and I do not know if it is possible. The closest I found was to take the pipeline.parameters and try to implement it manually, but it seems that there are some modules that are changed and are not the same as SkLearn. Besides, I'm sorry for my bad English.

**Comment by chukarsten** [comment]
@emanuelqueiroga, this is being addressed right now in #1956, which is currently in progress.  Standby for updates.

**Comment by emanuelqueiroga** [comment]
> @emanuelqueiroga , isso está sendo tratado agora no [# 1956](https://github.com/alteryx/evalml/issues/1956) , que está em andamento. Aguarde atualizações.

Ok, thank you.

**Comment by dsherry** [comment]
@emanuelqueiroga we just released v0.24.0 which contains a fix allowing the pipelines from `AutoMLSearch` to be pickled. Closing this issue for now, but please reach out if you have further trouble.
